src/commands/fal.rs

```diff
@@ -0,0 +1,215 @@
+                        }),
+                    }]],
+                })),
+            )
+            .await?;
+
+        ctx.bot_state.message_queue.wait_for_message(message.id).await?;
+        temp_file.close().unwrap();
+
+        Ok(())
+    }
+}
+
+impl Fal {
+    async fn generate(
+        &self,
+        ctx: &CommandContext,
+        prompt: String,
+    ) -> Result<Generation, CommandError> {
+        let request = FalRequest {
+            model_name: self.model_name,
+            prompt,
+            negative_prompt: String::new(),
+            image_size: ImageSize {
+                height: self.size.0,
+                width: self.size.1,
+            },
+            num_inference_steps: self.num_inference_steps,
+            guidance_scale: 5,
+            loras: vec![],
+            embeddings: vec![],
+            num_images: 1, // this variable is potentially changable - this is so cheap anyways
+            enable_safety_checker: true,
+            format: "jpeg",
+            safety_checker_version: "v1",
+        };
+
+        let response = ctx
+            .bot_state
+            .http_client
+            .post(format!("https://fal.run/fal-ai/{}", self.model_name))
+            .header("Authorization", format!("Key {}", env::var("FAL_API_KEY").unwrap()))
+            .json(&request)
+            .send()
+            .await?
+            .server_error()?;
+
+        if response.status() == StatusCode::OK {
+            let response = response.json::<FalResponse>().await?;
+            Ok(Generation {
+                image: response.images[0].clone(),
+            })
+        } else {
+            let response = response.json::<ErrorResponse>().await?;
+            Err(CommandError::Custom(response.error.message))
+        }
+    }
+}
+
+struct Generation {
+    image: String,
+}
+
+fn process_image(generation: &Generation) -> DynamicImage {
+    let image = image::load_from_memory_with_format(generation.image.as_bytes(), ImageFormat::Jpeg)
+        .unwrap();
+    image
+}
+
+fn format_result_text(generation: Generation) -> FormattedText {
+    let entities = vec!["generated ".text(), generation.image.text()];
+    formatted_text(entities)
+}
+
+#[derive(Serialize)]
+pub struct FalRequest {
+    pub model_name: &'static str,
+    pub prompt: String,
+    pub negative_prompt: String,
+    pub image_size: ImageSize,
+    pub num_inference_steps: u8,
+    pub guidance_scale: u8,
+    pub loras: Vec<String>,
+    pub embeddings: Vec<String>,
+    pub num_images: u8,
+    pub enable_safety_checker: bool,
+    pub format: &'static str,
+    pub safety_checker_version: &'static str,
+}
+
+#[derive(Serialize)]
+pub struct ImageSize {
+    pub height: u16,
+    pub width: u16,
+}
+
+#[derive(Deserialize)]
+pub struct FalResponse {
+    pub images: Vec<String>,
+}
+
+#[derive(Deserialize)]
+pub struct ErrorResponse {
+    pub error: Error,
+}
+
+#[derive(Deserialize)]
+pub struct Error {
+    // pub code: String,
```

Comment:
remove this comment
```suggestion
```

Reply:
literally why but ok

Reply:
you can also uncomment this field and put it inside error messages